sn_networking/src/record_store.rs

```diff
@@ -452,6 +435,32 @@ impl RecordStore for ClientRecordStore {
     fn remove_provider(&mut self, _key: &Key, _provider: &PeerId) {}
 }
 
+/// Cost calculator that increases cost nearing the maximum (2048).
+/// Table:
+///    1 =          0.000000010
+///    2 =          0.000000010
+///    4 =          0.000000011
+///    8 =          0.000000012
+///   16 =          0.000000014
+///   32 =          0.000000019
+///   64 =          0.000000036
+///  128 =          0.000000126
+///  256 =          0.000001591
+///  512 =          0.000253098
+/// 1024 =          6.405837009
+/// 2048 = 4103474778.282772064
+fn calculate_cost_at_step(step: usize) -> u64 {
+    assert!(step <= 2048, "step must be <= 2048");
+
+    // Using an exponential growth function: y = ab^x. Here, a is the starting cost and b is the growth factor.
+    // We want a function that starts with a low cost and only ramps up once we get closer to the maximum.
+    let a = 0.000_000_010_f64;
+    let b = 1.02_f64;
+    let y = a * b.powf(step as f64);
+
+    (y * 1_000_000_000_f64) as u64
```

Comment:
I'm not sure how this is tied to the TOTAL_SUPPLY? 

(And I think we should be dividing that supply by CLOSE_GROUP_SIZE to get a real cost to be charged by a given node, otherwise at MAX_RECORD_COUNT we'd be asking in total for `CLOSE_GROUP_SIZE*TOTAL_SUPPLY) ?

Reply:
It's not tied to the TOTAL_SUPPLY mathematically. I've picked these numbers by hand. According to the table, with MAX_RECORD_COUNT, the cost will be 4_103_474_778.282772064. I've only just realized by writing it out, that that's actually very close to 2^32. So indeed, probably too much. I can tweak the a and b variables a little bit to get to a lower maximum.

> otherwise at MAX_RECORD_COUNT we'd be asking in total for `CLOSE_GROUP_SIZE*TOTAL_SUPPLY)

I am not sure how this follows from the algorithm?

Reply:
each node will be asking for its own payment. So, at a worst case for the last remaining record, the total cost to send to all nodes would be CLOSE_GROUP_SIZE * COST. aka, (currently), 8 times the total supply. 

We can just divide TOTAL_SUPPLY by our CLOSE_GROUP_SIZE to get the actual max value any one node should be asking for?